File: breadth-first_search.py
#Relationship graph
graph = {}
graph['you'] = ['alice', 'bob', 'claire']
graph['alice'] = ['peggy']
graph['bob'] = ['anuj', 'peggy']
graph['claire'] = ['thom', 'jonny']
graph['anuj'] = []
graph['peggy'] = []
graph['thom'] = []
graph['jonny'] = []

#Let's get a deque (двусторонняя очередь)
from collections import deque

#Function to define whether person is a mango seller
def is_seller(person):
    if len(person) == 5:
        return True
    else:
        return False

#Function that will find the closest mango seller
def search(name):
    search_queue = deque()
    search_queue += graph[name]
    searched = []
    while search_queue:
        person = search_queue.popleft()
        if not person in searched:
            if is_seller(person):
                print(person + ' is a mango seller')
                return True
            else:
                search_queue += graph[person]
                searched.append(person)
    return False

search('you')

File: breadth-first_search.js
'use strict';

let deque = [];
let graph = {};
graph['you'] = ['Katya', 'Ksysha', 'Julia'];
graph['Katya'] = ['Nikitos'];
graph['Ksysha'] = ['Inna'];
graph['Julia'] = {};
graph['Julia']['friends'] = ['Denis'];
graph['Julia']['isSeller'] = true;
let person;
let checked = [];
function mangoSeller(name) {
    if (graph[person]['isSeller']) {
        return true;
    } else {
        return false;
    }
};

function search(name) {
    console.log(graph);
    deque += graph.name;
    console.log(deque);
    while (deque) {
        person = deque.shift();
        if (checked.indexOf(person) !== -1) {
            if (mangoSeller(person)) {
                console.log(person + 'is a mango seller');
            } else {
                deque += graph.peson;
                checked.append(person);
            }
        };
    };
};

// search('you');